tests/test_skills/test_registration_abci/test_behaviours.py

```diff
@@ -295,13 +296,20 @@ def mock_tendermint_update(self, valid_response: bool = True) -> None:
         response_kwargs = dict(status_code=200, body=body)
         self.mock_http_request(request_kwargs, response_kwargs)
 
-    def mock_tendermint_restart(self, valid_response: bool = True) -> None:
-        """Mock tendermint restart"""
-        url = self.state.tendermint_hard_reset_url
-        request_kwargs = dict(method="GET", url=url)
-        body = b"{}" if valid_response else b""
-        response_kwargs = dict(status_code=200, body=body)
-        self.mock_http_request(request_kwargs, response_kwargs)
+    @staticmethod
+    def dummy_reset_tendermint_with_wait_wrapper(
+        valid_response: bool,
+    ) -> Callable[[], Generator[None, None, Optional[bool]]]:
+        """Wrapper for a Dummy `reset_tendermint_with_wait` method."""
+
+        def dummy_reset_tendermint_with_wait(
+            **_: bool,
+        ) -> Generator[None, None, Optional[bool]]:
+            """Dummy `reset_tendermint_with_wait` method."""
+            yield
+            return valid_response
+
+        return dummy_reset_tendermint_with_wait
```

Comment:
ok seems like an excessive amount of code but if it works it works.

Reply:
I also do not like it, but could not think of another way. Let me know if you have any ideas.